src/database.ts

```diff
@@ -48,38 +131,40 @@ export class Database {
 		this.connection = mysql.createPool(config);
 		connectedDatabases.push(this);
 	}
-	query<T = ResultRow>(query: SQLStatement) {
-		const err = new Error();
+	resolveSQL(query: SQLStatement): [string, BasicSQLValue[]] {
+		let sql = query.sql[0];
+		const values = [];
+		for (let i = 0; i < query.values.length; i++) {
+			const value = query.values[i];
+			if (query.sql[i + 1].startsWith('`')) {
+				sql = sql.slice(0, -1) + this.connection.escapeId('' + value) + query.sql[i + 1].slice(1);
+			} else {
+				sql += '?' + query.sql[i + 1];
+				values.push(value);
+			}
+		}
+		return [sql, values];
+	}
+	query<T = ResultRow>(query: SQLStatement): Promise<T[]> {
```

Comment:
Consider making these template functions that work as `SQL`, instead of taking an `SQL` value, in my experience there will be a lot of `SQL` noise otherwise. There is no loss of expressivity as `query(x)` is equivalent to ``query`${x}` ``, and helpers that can take arguments can work as `helper(...args, x)` => ``helper(args)`${x}` ``.

Reply:
This works for the basic `query` functions, but helpers aren't quite so easy because the raw part is generally optional...